### DISCUSSION TITLE
Deactivate eglot features/activities to speed up performance

### DISCUSSION THREAD

**Discussion by buhtz** [joaotavora/eglot#947] [General]
LSP, eglot and Emacs are quit new to me. Everything is still magic. ;)

Beside a usual desktop PC I use Emacs also on a headless RaspberryPi 4 (via SSH terminal).
Using eglot (or any other lsp mode) use a lot of CPU resources on the Pi.

I would like to know how could I modify eglot (in Python mode) itself or the underlying python-lsp-server to use less CPU.

For example can I increase the interval when current buffer is PEP8-checked? Maybe I can deactivate it complete and only run it explicite via a command or keyboard shortcut?

Or can I deactivate the look-into-and-show-me-the-current-functions-docstring-in-minibuffer thing?

But I won't lose the auto-completion feature.

Are there any other suggestion what could be deactivated?

**Comment by joaotavora** [comment]
@Codeberg-AsGithubAlternative-buhtz I think the answer to all those questions is yes, but I don't have much time to go into details:

> For example can I increase the interval when current buffer is PEP8-checked?

Yes, see `flymake-no-changes-timeout`

> Maybe I can deactivate it complete and only run it explicite via a command or keyboard shortcut?

I think so, read the flymake manual.

> Or can I deactivate the look-into-and-show-me-the-current-functions-docstring-in-minibuffer thing?

See `eglot-ignored-server-capabilities`.

> Are there any other suggestion what could be deactivated?

In general, read Eglot's README.  It's not great, but it helps.
